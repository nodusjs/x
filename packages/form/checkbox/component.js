import { html } from "@nodusjs/std/dom";

export const component = (textarea) => html`
  <input
    type="checkbox"
    ${textarea.id ? `id="${textarea.id}"` : ""}
    ${textarea.name ? `name="${textarea.name}"` : ""}
    ${textarea.checked ? "checked" : ""}
    ${textarea.disabled ? "disabled" : ""}
    ${textarea.readonly ? "readonly" : ""}
    ${textarea.required ? "required" : ""}
  />
  <wrap>
    <label for="${textarea.id}">
      <slot name="label"></slot>
    </label>
    <slot name="validity"></slot>
  </wrap>
`;
