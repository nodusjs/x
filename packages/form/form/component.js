import { html } from "@nodusjs/std/dom";

export const component = (form) =>
  html`
    <form>
      ${form.template}
    </form>
  `;
