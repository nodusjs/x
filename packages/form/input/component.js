import { html } from "@nodusjs/std/dom";

export const component = (input) => html`
  <label for="${input.id}">
    <slot name="label"></slot>
  </label>
  <input
    ${input.id ? `id="${input.id}"` : ""}
    ${input.inputMode ? `inputmode="${input.inputMode}"` : ""}
    ${input.max ? `max="${input.max}"` : ""}
    ${input.maxLength ? `maxlength="${input.maxLength}"` : ""}
    ${input.min ? `min="${input.min}"` : ""}
    ${input.minLength ? `minlength="${input.minLength}"` : ""}
    ${input.name ? `name="${input.name}"` : ""}
    ${input.placeholder ? `placeholder="${input.placeholder}"` : ""}
    ${input.type ? `type="${input.type}"` : ""}
    ${input.value ? `value="${input.value}"` : ""}
    ${input.readonly ? "readonly" : ""}
    ${input.required ? "required" : ""}
  />
  <slot name="validity"></slot>
`;
