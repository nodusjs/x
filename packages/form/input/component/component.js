import { html } from "@nodusjs/std/dom";

/**
 * Gera o template de renderização para o `<x-input>`, incluindo o rótulo,
 * o elemento `<input>` com seus atributos refletidos e um slot para mensagens de validade.
 *
 * @param {import("./input").default} input Instância do componente `<x-input>` cujas propriedades
 * serão usadas para definir os atributos do elemento `<input>`.
 * @returns {import("@nodusjs/std/dom").TemplateResult} Fragmento de template pronto para injeção
 * no Shadow DOM do componente.
 *
 * @description
 * Esta função monta o HTML do `<x-input>` usando a API de template do `@nodusjs/std/dom`.
 * Ela garante que os atributos (como `id`, `name`, `type`, `value`, `max`, `min`, `readonly`,
 * `required`, etc.) estejam sempre sincronizados com as propriedades do componente, além de
 * fornecer um `<slot name="validity">` para exibição de mensagens de validação.
 *
 * @example
 * import Input from "./input.js";
 * import { component as inputTemplate } from "./component.js";
 *
 * const inp = new Input();
 * inp.id = "email";
 * inp.type = "email";
 * inp.placeholder = "Digite seu e-mail";
 *
 * // Em um contêiner temporário:
 * const container = document.createElement("div");
 * container.innerHTML = inputTemplate(inp);
 * console.log(container.innerHTML);
 * // → '<label for="email"><slot name="label"></slot></label>'
 * //    '<input id="email" type="email" placeholder="Digite seu e-mail">'
 * //    '<slot name="validity"></slot>'
 */
export const component = (input) => html`
  <label for="${input.id}">
    <slot name="label"></slot>
  </label>
  <input
    ${input.id ? `id="${input.id}"` : ""}
    ${input.inputMode ? `inputmode="${input.inputMode}"` : ""}
    ${input.max ? `max="${input.max}"` : ""}
    ${input.maxLength ? `maxlength="${input.maxLength}"` : ""}
    ${input.min ? `min="${input.min}"` : ""}
    ${input.minLength ? `minlength="${input.minLength}"` : ""}
    ${input.name ? `name="${input.name}"` : ""}
    ${input.type ? `type="${input.type}"` : ""}
    ${input.value ? `value="${input.value}"` : ""}
    ${input.readonly ? "readonly" : ""}
    ${input.required ? "required" : ""}
  />
  <slot name="validity"></slot>
`;
