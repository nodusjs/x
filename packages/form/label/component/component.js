import { html } from "@nodusjs/std/dom";

/**
 * Gera o template HTML para o conteúdo de `<x-label>`, projetando tudo que
 * for colocado dentro da tag no seu próprio `<slot>` no Shadow DOM.
 *
 * @returns {DocumentFragment} Fragmento contendo `<slot></slot>` para projeção do conteúdo filho.
 *
 * @example
 * import { component } from "./component.js";
 *
 * // Em um Custom Element definido com render(component):
 * const tpl = component();
 * console.log(tpl.firstChild.outerHTML);
 * // => "<slot></slot>"
 */
export const component = () => html`<slot></slot>`;
