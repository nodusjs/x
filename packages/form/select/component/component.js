import { html } from "@nodusjs/std/dom";

export const component = (select) => html`
  <label for="${select.id}">
    <slot name="label"></slot>
  </label>
  <wrap>
    <select
      ${select.id ? `id="${select.id}"` : ""}
      ${select.name ? `name="${select.name}"` : ""}
      ${select.value ? `value="${select.value}"` : ""}
      ${select.readonly ? "readonly" : ""}
      ${select.required ? "required" : ""}
    />
      <option></option>
    </select>
    <x-icon use="keyboard_arrow_down"></x-icon>
  </wrap>
  <slot name="validity"></slot>
`;
