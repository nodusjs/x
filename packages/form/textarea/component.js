import { html } from "@nodusjs/std/dom";

export const component = (textarea) => html`
  <label for="${textarea.id}">
    <slot name="label"></slot>
  </label>
  <textarea
    ${textarea.id ? `id="${textarea.id}"` : ""}
    ${textarea.name ? `name="${textarea.name}"` : ""}
    ${textarea.value ? `value="${textarea.value}"` : ""}
    ${textarea.readonly ? "readonly" : ""}
    ${textarea.required ? "required" : ""}>${textarea.value}</textarea>
  <slot name="validity"></slot>
`;
