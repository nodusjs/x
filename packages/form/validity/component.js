import { html } from "@nodusjs/std/dom";

/**
 * Gera o template de slot para exibir mensagens de validação.
 *
 * @param {void} _ - Nenhum parâmetro é necessário para este template.
 * @returns {DocumentFragment} Fragmento contendo o slot nomeado "validity".
 *
 * @description
 * Cria um slot nomeado que permite projetar mensagens de erro ou feedback de
 * validação dentro do campo personalizado `<x-input>`. Esse slot deve ser
 * preenchido via light DOM pelo componente pai para exibir o texto de validade.
 *
 * @example
 * import { component } from "./component.js";
 *
 * const fragment = component();
 * const container = document.createElement("div");
 * container.appendChild(fragment.cloneNode(true));
 *
 * console.log(container.innerHTML);
 * // => '<slot></slot>'
 */
export const component = () => html`<slot></slot>`;
