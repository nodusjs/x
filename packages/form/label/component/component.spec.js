import { describe, expect, it } from "vitest";
import { component } from "./component.js";

describe("component", () => {
  it("deve gerar exatamente um <slot> para projeção do conteúdo filho", () => {
    const html = component();
    expect(html).toMatchSnapshot();
  });
});
