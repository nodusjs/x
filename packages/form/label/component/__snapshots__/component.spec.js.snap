// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component > deve gerar exatamente um <slot> para projeção do conteúdo filho 1`] = `"<slot></slot>"`;
